src/main/java/org/osiam/resources/scim/Resource.java

```diff
@@ -41,27 +40,16 @@
     private final String id;
     private final String externalId;
     private final Meta meta;
-    @JsonProperty(required = true)
     private final Set<String> schemas;
 
-    protected Resource(@JsonProperty("id") String id,
-                       @JsonProperty("externalId") String externalId,
-                       @JsonProperty("meta") Meta meta,
-                       @JsonProperty(value = "schemas", required = true) Set<String> schemas) {
+    protected Resource(String id, String externalId, Meta meta, Set<String> schemas) {
         this.id = id;
         this.externalId = externalId;
         this.meta = meta;
         if (schemas == null || schemas.isEmpty()) {
             throw new SCIMDataValidationException("Schemas cannot be null or empty!");
         }
-        this.schemas = schemas;
-    }
-
-    protected Resource(Builder builder) {
```

Comment:
I don't like the idea, that the object, that has to be built, needs to know anything about its builders. From the object's point of view, it simply does not care, who builds it. The object just provides a full primary constructor, that clients like our builder (or Jackson) can use. In the end, the builder only adds syntactic sugar, so that creating a new user is more convenient and less error prone. But one could also just call the constructor (if it were publicly visible).

It also tightly couples the object to its builder, which requires a change to the object, if we want to use another builder class. And having another builder class might actually become true in the near future, when we further revise the API and get rid of the inner classes, for example.

Speaking of which, we made the builders inner classes of the objects, which already breaks encapsulation. This finally led to issues like the one addressed in #206. I don't think, that the builder is supposed to access the internal, private state of the object and vice versa. So, your solution would actually require to add getters to the builder, so that the constructor of the object can consume the values set on the builder.

BTW: I think, that having a constructor that takes the builder as an argument is less cohesive than letting the builder just call the primary constructor, because we are not re-using the already existent primary constructor, but add a new one, possibly duplicating code.